src/lib/mongoBackend/MongoGlobal.h

```diff
@@ -133,9 +133,15 @@ using namespace mongo;
 /*****************************************************************************
 *
 * Macros to ease extracting fields from BSON objects
+*
+* FIXME P10: probably the STR_FIELD macro (and maybe C_STR_FIELD) are not needed at
+* the end
 */
-#define STR_FIELD(i, sf) std::string(i.getStringField(sf))
-#define C_STR_FIELD(i, sf) i.getStringField(sf)
+//#define STR_FIELD(i, sf) std::string(i.getStringField(sf))
+//#define C_STR_FIELD(i, sf) i.getStringField(sf)
+#include "mongoBackend/safeBsonGet.h"
+#define STR_FIELD(i, sf) getStringField(i, sf)
+#define C_STR_FIELD(i, sf) getStringField(i, sf).c_str()
```

Comment:
Yeah, I agree.
A definition to just change the name of the 'function' doesn't help much. Just makes it more confusing.
I would definitely vote for *not* using STR_FIELD and C_STR_FIELD and instead use the functions themselves